packages/editor/src/lib/utils/sync/LocalIndexedDb.ts

```diff
@@ -42,9 +42,12 @@ async function openLocalDb(persistenceKey: string) {
 }
 
 async function migrateLegacyAssetDbIfNeeded(persistenceKey: string) {
-	const databases = await window.indexedDB.databases()
+	let databases = (await window.indexedDB.databases?.())?.map((db) => db.name)
+	if (!databases) {
```

Comment:
Don't think we'll ever end up in here. databases is `(string | undefined)[]`, which is always truthy.
```suggestion
	if (!databases.length) {
```

Reply:
Or maybe even add a check for databases
```typescript
if(!window.indexedDB.databases)
```

Reply:
I think vscode is showing the wrong type info here actually:
`databases` is actually: `string[] | undefined`

but i'm happy to make this clearer!

Reply:
Yeah. My guess is that it can't understand that `databases` might not exist since it's checking the latest version of the api or something like that?

In any case I think the change you made is much easier to read and also removes any ambiguities.